### PR TITLE
Use ClimaCore operators in compute_covariance_shear

### PR DESCRIPTION
**Pull request by charleskawczynski**

### CODE REVIEW

File: src/Turbulence_PrognosticTKE.jl [hunk 1]
```diff
@@ -605,34 +603,18 @@ function compute_covariance_shear(
...
-
-            u_cut = ccut(prog_gm.u, grid, k)
-            ∇u = c∇(u_cut, grid, k; bottom = Extrapolate(), top = SetGradient(0))
-
-            var2_dual = dual_faces(EnvVar2, grid, k)
-            var1_dual = dual_faces(EnvVar1, grid, k)
-
-            ∇var2 = ∇f2c(var2_dual, grid, k; bottom = SetValue(0), top = SetGradient(0))
-            ∇var1 = ∇f2c(var1_dual, grid, k; bottom = SetValue(0), top = SetGradient(0))
+    bcs = (; bottom = CCO.Extrapolate(), top = CCO.SetGradient(0))
```
Comment (simonbyrne) [inline src/Turbulence_PrognosticTKE.jl:606]:
```suggestion
    bcs = (; bottom = CCO.Extrapolate(), top = CCO.SetGradient(Geometry.WVector(0))
```
a gradient of a scalar should be a vector

Comment (charleskawczynski) [reply src/Turbulence_PrognosticTKE.jl:606]: I'll try that

Comment (charleskawczynski) [reply src/Turbulence_PrognosticTKE.jl:606]: Hmm, I'm still getting the same error with that..

Comment (simonbyrne) [reply src/Turbulence_PrognosticTKE.jl:606]: exactly the same error? can you paste the stacktrace?

Comment (charleskawczynski) [reply src/Turbulence_PrognosticTKE.jl:606]: > exactly the same error?
I think so? I'm not 100% sure

> can you paste the stacktrace?
Not practically, the stack traces are too long. I'll push a commit that adds your suggestion

Comment (charleskawczynski) [reply src/Turbulence_PrognosticTKE.jl]: With your suggestion, the truncated stack trace is

```
RROR: LoadError: MethodError: no method matching Jcontravariant3(::Float64, ::ClimaCore.Geometry.LocalGeometry{ClimaCore.Geometry.ZPoint{Float64}, Float64, ClimaCore.Geometry.AxisTensor{Float64, 2, Tuple{ClimaCore.Geometry.CartesianAxis{(3,)}, ClimaCore.Geometry.CovariantAxis{(3,)}}, StaticArrays.SMatrix{1, 1, Float64, 1}}, ClimaCore.Geometry.AxisTensor{Float64, 2, Tuple{ClimaCore.Geometry.ContravariantAxis{(3,)}, ClimaCore.Geometry.CartesianAxis{(3,)}}, StaticArrays.SMatrix{1, 1, Float64, 1}}})
--
  | Closest candidates are:
  | Jcontravariant3(::ClimaCore.Geometry.AxisTensor, ::ClimaCore.Geometry.LocalGeometry) at /central/scratch/climaci/turbulenceconvection-ci/depot/cpu/packages/ClimaCore/Xu0dF/src/Geometry/conversions.jl:69
  | Stacktrace:
  | [1] stencil_interior(#unused#::ClimaCore.Operators.DivergenceF2C{NamedTuple{(), Tuple{}}}, loc::ClimaCore.Operators.LeftBoundaryWindow{:bottom}, idx::Int64, arg::ClimaCore.Fields.Field{Cl
```

Comment (charleskawczynski) [reply src/Turbulence_PrognosticTKE.jl]: Looks like we had to wrap other parts of the expression with Cartesian3Vector. Thanks for the pointer!

**Comment by charleskawczynski** [comment]
Results are virtually unchanged.

**Pull request merged**